inference-engine/src/mkldnn_plugin/nodes/mkldnn_roi_align_node.cpp

```diff
@@ -0,0 +1,389 @@
+        ctx.node.executeSpecified<srcT, dstT>(ctx.strm);
+    }
+};
+void MKLDNNROIAlignNode::execute(mkldnn::stream strm) {
+    auto inputPrec = getParentEdgeAt(0)->getMemory().GetDescriptor().data.data_type;
+    auto outputPrec = getChildEdgeAt(0)->getMemory().GetDescriptor().data.data_type;
+    if (!((inputPrec == mkldnn_bf16 && outputPrec == mkldnn_bf16) ||
+          (inputPrec == mkldnn_f32 && outputPrec == mkldnn_f32)))
+        THROW_IE_EXCEPTION <<"ROIAlign doesn't support demanded precisions";
+
+    ROIAlignContext ctx = {
+            *this,
+            strm
+    };
+
+    OV_SWITCH(MKLDNNPlugin, ROIAlignExecute, ctx, std::tie(inputPrec, outputPrec),
+              OV_CASE2(mkldnn_f32, mkldnn_f32, float, float),
+              OV_CASE2(mkldnn_bf16, mkldnn_bf16, bfloat16_t, bfloat16_t))
+}
+
+template <typename inputType, typename outputType>
+void MKLDNNROIAlignNode::executeSpecified(mkldnn::stream strm) {
+    auto &srcMemory0 = getParentEdgeAt(0)->getMemory();
+    auto &srcMemory1 = getParentEdgeAt(1)->getMemory();
+    auto &srcMemory2 = getParentEdgeAt(2)->getMemory();
+
+    auto &dstMemory = getChildEdgeAt(0)->getMemory();
+    int blockSize;
+    auto selectedFmt = srcMemory0.GetDescriptor().data.format;
+    switch (selectedFmt) {
+        case mkldnn_nchw:
+        case mkldnn_nhwc:
+            blockSize = 1;
+            break;
+        case mkldnn_nChw16c:
+            blockSize = 16;
+            break;
+        case mkldnn_nChw8c:
+            blockSize = 8;
+            break;
+        default:
+            THROW_IE_EXCEPTION << "unsupported format for ROIAlign";
+    }
+
+    const auto *srcData = reinterpret_cast<const inputType *>(srcMemory0.GetData());
+    const auto *srcRoi = reinterpret_cast<const float *>(srcMemory1.GetData());
+    const auto *srcRoiIdx = reinterpret_cast<const int *>(srcMemory2.GetData());
+    auto *dst = reinterpret_cast<outputType *>(dstMemory.GetData());
+
+    auto nominalRoiCount = static_cast<int>(srcMemory1.GetDims()[0]);
+    int realRois = 0;
+    int C = static_cast<int>(srcMemory0.GetDims()[1]);
+    int H = static_cast<int>(srcMemory0.GetDims()[2]);
+    int W = static_cast<int>(srcMemory0.GetDims()[3]);
+
+    const int binCount = pooledH * pooledW;
+
+    int hInputCoeff;
+    int wInputCoeff;
+    int hOutputCoeff;
+    int wOutputCoeff;
+    if (selectedFmt == mkldnn_nhwc) {
+        hInputCoeff = W * C;
+        wInputCoeff = C;
+        hOutputCoeff = pooledW * C;
+        wOutputCoeff = C;
+    } else {  // nchw, nChw16c, nChw8c
+        hInputCoeff = W * blockSize;
+        wInputCoeff = blockSize;
+        hOutputCoeff = pooledW * blockSize;
+        wOutputCoeff = blockSize;
+    }
+
+    for (; realRois < nominalRoiCount; realRois++) {
+        const int *srcRoiIdxPtr = &srcRoiIdx[realRois];
+        auto roiBatchInd = srcRoiIdxPtr[0];
+        if (roiBatchInd == -1) {
+            break;
+        }
+    }
+
+    int roiOff;
```

Comment:
this variable not used anywhere else cycle and previous values of thus values too
lets move into cycle body

Reply:
Done